### ISSUE TITLE
Unable to install on vSphere with a datastore containing a space in the name

### ISSUE THREAD

**Issue by bodgit** [coreos/tectonic-installer#3051]
## What keywords did you search in tectonic-installer issues before filing this one?

vmware

## Is this a BUG REPORT or FEATURE REQUEST?

BUG REPORT

### Tectonic version ([release](https://github.com/coreos/tectonic-installer/releases) or commit hash):
```
1.8.7-tectonic.1
```

### Terraform version (`terraform version`):

```
0.10.8
```
### Platform (aws|azure|openstack|metal|vmware):

```
vmware
```

### What happened?

I get the following errors when trying to install Tectonic on vSphere:
```
Error: Error refreshing state: 3 error(s) occurred:

* module.workers.vsphere_virtual_machine.node: 2 error(s) occurred:

* module.workers.vsphere_virtual_machine.node[0]: vsphere_virtual_machine.node.0: [ERROR] Failed trying to parse disk path: [Pure Labs] mycluster-worker-0/mycluster-worker-0.vmdk
* module.workers.vsphere_virtual_machine.node[1]: vsphere_virtual_machine.node.1: [ERROR] Failed trying to parse disk path: [Pure Labs] mycluster-worker-1/mycluster-worker-1.vmdk
* module.masters.vsphere_virtual_machine.node: 2 error(s) occurred:

* module.masters.vsphere_virtual_machine.node[1]: vsphere_virtual_machine.node.1: [ERROR] Failed trying to parse disk path: [Pure Labs] mycluster-master-1/mycluster-master-1.vmdk
* module.masters.vsphere_virtual_machine.node[0]: vsphere_virtual_machine.node.0: [ERROR] Failed trying to parse disk path: [Pure Labs] mycluster-master-0/mycluster-master-0.vmdk
* module.etcd.vsphere_virtual_machine.etcd_node: 3 error(s) occurred:

* module.etcd.vsphere_virtual_machine.etcd_node[0]: vsphere_virtual_machine.etcd_node.0: [ERROR] Failed trying to parse disk path: [Pure Labs] mycluster-etcd-0/mycluster-etcd-0.vmdk
* module.etcd.vsphere_virtual_machine.etcd_node[1]: vsphere_virtual_machine.etcd_node.1: [ERROR] Failed trying to parse disk path: [Pure Labs] mycluster-etcd-1/mycluster-etcd-1.vmdk
* module.etcd.vsphere_virtual_machine.etcd_node[2]: vsphere_virtual_machine.etcd_node.2: [ERROR] Failed trying to parse disk path: [Pure Labs] mycluster-etcd-2/mycluster-etcd-2.vmdk
```
Searching for the above error suggests that this is caused by terraform-providers/terraform-provider-vsphere#213 (and terraform-providers/terraform-provider-vsphere#52) which is caused by the datasource name containing a space. Looking at the tectonic-installer manifests shows it's pinned to version 0.2.2 of the vSphere provider and that 1.1.1 should fix this. However I can see that you've already tried to upgrade to 0.4.2 and reverted it again in #2564.

### What you expected to happen?

```
I was expecting the install to complete.
```

### How to reproduce it (as minimally and precisely as possible)?

```
Have a vSphere datastore with a space in the name, it would seem.
```

### References

```
Only issues/PRs external to this project, see the above referenced issues.
```

**Comment by bodgit** [comment]
For now, I've built my own patched version 0.2.2 of the vsphere provider with the following diff applied:
```diff
diff --git a/vsphere/resource_vsphere_virtual_machine.go b/vsphere/resource_vsphere_virtual_machine.go
index c5a74a8..17e9427 100644
--- a/vsphere/resource_vsphere_virtual_machine.go
+++ b/vsphere/resource_vsphere_virtual_machine.go
@@ -4,6 +4,7 @@ import (
        "fmt"
        "log"
        "net"
+       "regexp"
        "strconv"
        "strings"

@@ -1026,7 +1027,16 @@ func resourceVSphereVirtualMachineRead(d *schema.ResourceData, meta interface{})
                        log.Printf("[DEBUG] resourceVSphereVirtualMachineRead - Analyzing disk: %v", diskFullPath)

                        // Separate datastore and path
-                       diskFullPathSplit := strings.Split(diskFullPath, " ")
+                       /*
+                               diskFullPathSplit := strings.Split(diskFullPath, " ")
+                       */
+                       s := regexp.MustCompile(`^(\[[^]]+\])\s(.+)$`).FindAllStringSubmatch(diskFullPath, -1)
+                       if s == nil {
+                               return fmt.Errorf("[ERROR] Failed trying to parse disk path: %v", diskFullPath)
+                       }
+
+                       diskFullPathSplit := s[0][1:]
+
                        if len(diskFullPathSplit) != 2 {
                                return fmt.Errorf("[ERROR] Failed trying to parse disk path: %v", diskFullPath)
                        }
```
This allows the installer to continue without error (although I now hit a different, unrelated problem), but ultimately a newer version of the provider should be validated and used.

**Comment by bodgit** [comment]
Closing as this isn't likely to ever get fixed.
